docs/en/advanced_tutorials/data_element.md

```diff
@@ -1,3 +1,1096 @@
+
+```python
+data_element = BaseDataElement(
+    bboxes=torch.rand((5, 4)),
+    scores=torch.rand((5,)),
+    metainfo=dict(img_id=1, img_shape=(640, 640)))
+
+# set metainfo and data while creating BaseDataElement, makes BaseDataElement has the same unused data
+data_element1 = data_element.new(metainfo=dict(img_id=2, img_shape=(320, 320)))
+print('bboxes is in data_element1:', 'bboxes' in data_element1) # True
+print('bboxes in data_element1 is same as bbox in data_element', (data_element1.bboxes == data_element.bboxes).all())
+print('img_id in data_element1 is', data_element1.img_id == 2) # True
+
+data_element2 = data_element.new(label=torch.rand(5,))
+print('bboxes is not in data_element2', 'bboxes' not in data_element2) # True
+print('img_id in data_element2 is same as img_id in data_element', data_element2.img_id == data_element.img_id)
+print('label in data_element2 is', 'label' in data_element2)
+
+# create a new object using `clone`, which makes the new object has the same data, same metainfo, and the same status as the data_element
+data_element2 = data_element1.clone()
+```
+
+```
+bboxes is in data_element1: True
+bboxes in data_element1 is same as bbox in data_element tensor(True)
+img_id in data_element1 is True
+bboxes is not in data_element2 True
+img_id in data_element2 is same as img_id in data_element True
+label in data_element2 is True
+```
+
+### 3. Add and query attributes
+
+For adding `data` attributes, users can add them as adding class attributes inside `data`. However, for `metainfo`, users should use `set_metainfo` interface to explicitly modify the information if it is needed since usually `metainfo` contains the metadata of the images and will not be modify in most scenarios.
+
+For querying, users can access keys that exists only in `data` via `keys`, `values`, and `items`. You can also use `metainfo_keys`, `metainfo_values`, and `metainfo_items` to access those only exists in `metainfo`.
+
+Moreover, for the ease of use, users can access `data` and `metainfo` as if they were working with class properties, or through the `get()` interface as class dictionaries.
+
+**Note:**
+
+1. `BaseDataElement` does not support fields with the same name in `metainfo` and `data` attributes, which makes users should avoid setting the same fields which already existed. Otherwise `BaseDataElement` will throw out an error.
+2. Considering the `InstanceData` and `PixelData` support data slicing, and to avoid the inconsistent use of `[]` and to reduce different methods for the same requirement, `BaseDataElement` does not support accessing and setting its properties like a dictionary. Therefore, you cannot use operations like `BaseDataElement[name]`.
```

Comment:
> Refine this description

How about changing this part to:

1. `BaseDataElement` does not allow users to have fields using the same names in both `metainfo` and `data`. Otherwise, it will throw out an error.
2. Considering that both `InstanceData` and `PixelData` support slicing operation, to avoid inconsistency in the use of `[]` and to reduce different methods for the same needs, `BaseDataElement` does not allow accessing and setting values like a dictionary. Therefore, operations like `BaseDataElement[ name]` are not supported.

Reply:
I think this presentation `and to reduce different methods for the same needs, BaseDataElement ` is not necessary, both for Chinese docs.

Reply:
I think you can directly remove the second description. It somehow conflicts with the current implementation

Reply:
> I think this presentation `and to reduce different methods for the same needs, BaseDataElement ` is not necessary, both for Chinese docs.

Agreed.

Then, how about:

1. `BaseDataElement` does not allow users to use the same name to set fields in `metainfo` and `data`. Otherwise, it will throw out an error.
1. Considering that both `InstanceData` and `PixelData` support the slicing operation and to avoid inconsistency in the use of `[]`, `BaseDataElement` does not allow accessing and setting values like a dictionary.